### PR TITLE
[SecurityBundle] Don't require a user provider for the anonymous listener

### LINKED ISSUE #34504: [Security] Firewall requires the `provider` to be configured even if it does not allow full authentication
**Symfony version(s) affected**: 4.4.0

**Description**  
Hello.

Next error from 4.3 to 4.4. My monolith throws an exception after upgrading to 4.4 about security.

`Not configuring explicitly the provider for the "anonymous" listener on "main" firewall is ambiguous as there is more than one registered provider.`

In SF 4.3 everything works fine. I have not noticed any changes in this topic in the history of changes.

**How to reproduce**  
My security.yaml looks like:

```
security:
    providers:
        intranet_users:
            entity: { class: SOME\ENTITY\CLASS, property: email }
        rest_api_provider:
            memory:
                users:
                    SOME_REST_LOGIN: { password: SOME_REST_PASSWORD, roles: 'ROLE_REST_API_USER' }
    
    firewalls:
        dev:
            pattern:  ^/(_(profiler|wdt)|css|images|js)/
            security: false
        api:
            pattern:   ^/(webservices/rest/)
            provider:  rest_api_provider
            anonymous: false
            http_basic:
                realm: 'SOME DESCRIPTION'
        main:
            anonymous: ~

```

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #34504 
| License       | MIT
| Doc PR        | -

Forgotten when adding the AnonymousFactory in #33503

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php [hunk 1]
```diff
@@ -464,7 +464,7 @@ private function createAuthenticationListeners(ContainerBuilder $container, stri
                             throw new InvalidConfigurationException(sprintf('Invalid firewall "%s": user provider "%s" not found.', $id, $firewall[$key]['provider']));
                         }
                         $userProvider = $providerIds[$normalizedName];
-                    } elseif ('remember_me' === $key) {
+                    } elseif ('remember_me' === $key || 'anonymous' === $key) {
                         // RememberMeFactory will use the firewall secret when created
```
Comment (xabbuh) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: Should we update this comment?

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php (hunk 1 above)
Comment (chalasr) [reply src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: done

Comment (chalasr) [reply src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: done

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**